sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -69,6 +69,7 @@ trait BaseLimitExec extends UnaryExecNode with CodegenSupport {
   override def doConsume(ctx: CodegenContext, input: Seq[ExprCode], row: ExprCode): String = {
     val stopEarly = ctx.freshName("stopEarly")
     ctx.addMutableState("boolean", stopEarly, s"$stopEarly = false;")
+    shouldStopRequired = true // loop may break early even without append in loop body
```

Comment:
Good catch. This implementation depends on slightly old revision that means there is no `stopEarly()` method. Removed this line.